tests/test_locality_NeighborQuery.py

```diff
@@ -504,6 +504,16 @@ def test_duplicate_cell_shells2(self):
         q = nq.query(positions[0], dict(num_neighbors=1000))
         self.assertEqual(len(list(q)), 3)
 
+    def test_2d_box_3d_points(self):
+        """Test that points with z != 0 fail if the box is 2D."""
+        L = 10  # Box Dimensions
+        r_max = 2.01  # Cutoff radius
+
+        box = freud.box.Box.square(L)  # Initialize Box
+        points = np.array([[0, 0, 0], [0, 1, 1]])
+        with self.assertRaises(ValueError):
+            self.build_query_object(box, points, r_max)
+
```

Comment:
Can you also add a test to some 2D method (e.g. PMFTXY2D) and make sure it fails too? Ideally it seems like something to test more systematically but until then at least a spot check or two would be good.